Makefile

```diff
@@ -1263,44 +1276,175 @@ checkstack: tools/checkstack$(EXE)
 
 .INTERMEDIATE: tools/checkstack$(EXE) tools/checkstack.$(O)
 tools/checkstack$(EXE): tools/checkstack.$(O)
-	$(MAKE) -C tools checkstack$(EXE)
+	$(MKEXE) $(OUTPUTEXE)$@ $<
 else
 checkstack:
 	@
 endif
 
 # Lint @since and @deprecated annotations
 
+lintapidiff_LIBRARIES = \
+  $(addprefix compilerlibs/ocaml,common bytecomp middleend) \
+  otherlibs/str/str
+lintapidiff_MODULES = tools/lintapidiff
+
+tools/lintapidiff.opt$(EXE): VPATH += $(ROOTDIR)/otherlibs/str
```

Comment:
I think this one doesn't need `$(ROOTDIR)`. It looks like there a few tweaks needed to be able to say `make lintapidiff` -  5f2f478d8aee8fdb470d2dd5a4efa06be753cb86 seemed to be working for me (at least to get it to build!)